imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,94 @@
+""" Read/Write images using rawpy.
+
+rawpy is an easy-to-use Python wrapper for the LibRaw library. 
+It also contains some extra functionality for finding and repairing hot/dead pixels.
+"""
+
+import rawpy
+import numpy as np
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union, cast
+from ..core.request import URI_BYTES, InitializationError, IOMode, Request
+from ..core.v3_plugin_api import PluginV3
+from ..typing import ArrayLike
+
+
+class RawPyPlugin(PluginV3):
+    """A class representing the rawpy plugin.
+
+    Methods
+    -------
+
+    .. autosummary::
+    :toctree: _plugins/rawpy
+
+    RawPyPlugin.read
+    """
+    def __init__(self, request: Request) -> None:
+        """Instantiates a new rawpy plugin object
+
+        Parameters
+        ----------
+        request: {Request}
+            A request object representing the resource to be operated on.
+        """
+
+        super().__init__(request)
+
+        self._image_file = None
+
+        if request.mode.io_mode == IOMode.read:
+            try:
+                with rawpy.RawPy.open_buffer(request.get_file()):
+                    self._image_file = request.get_file()
```

Comment:
Seeing this now, don't forget to also implement the `def close()` method to close the file once we are done with it and avoid leaking a file handle.

Reply:
Thank you for your comment. I have updated it now accordingly.